### ISSUE TITLE
Reduce size of package

### ISSUE THREAD

**Issue by bkniffler** [FormidableLabs/victory#202]
I've replaced charts.js with victory. This has resulted in an increase of 250kb on my optimized bundle.js file sie, from 750kb to 1mb. Thats an insane amount for one library, especially since its a replacement, not an addition!

Here is the result of the webpack stats analysis

```
victory: 2.59 MB (39.8%)
  victory-axis: 480.61 KB (18.1%)
    victory-util: 207.91 KB (43.3%)
      d3-scale: 141.63 KB (68.1%)
        d3-time-format: 27.13 KB (19.2%)
          <self>: 27.13 KB (100%)
        d3-time-format/build/d3-time-format.js|...: 27.13 KB (19.2%)
          <self>: 0 B (0.00%)
        d3-format: 14.26 KB (10.1%)
          <self>: 14.26 KB (100%)
        d3-color: 13.5 KB (9.53%)
          <self>: 13.5 KB (100%)
        d3-arrays: 12.66 KB (8.94%)
          <self>: 12.66 KB (100%)
        d3-interpolate: 11.98 KB (8.46%)
          <self>: 11.98 KB (100%)
        d3-time: 10.38 KB (7.33%)
          <self>: 10.38 KB (100%)
        <self>: 24.59 KB (17.4%)
      d3-scale/build/d3-scale.js|...: 24.59 KB (11.8%)
        <self>: 0 B (0.00%)
      reduce-css-calc: 7.88 KB (3.79%)
        reduce-function-call: 1.86 KB (23.6%)
          <self>: 1.86 KB (100%)
        reduce-function-call/index.js|...: 1.86 KB (23.6%)
          <self>: 0 B (0.00%)
        balanced-match: 554 B (6.87%)
          <self>: 554 B (100%)
        <self>: 3.61 KB (45.9%)
      reduce-css-calc/index.js|...: 3.61 KB (1.74%)
        <self>: 0 B (0.00%)
      <self>: 30.2 KB (14.5%)
    radium: 129.24 KB (26.9%)
      inline-style-prefixer: 50.82 KB (39.3%)
        bowser: 8.52 KB (16.8%)
          <self>: 8.52 KB (100%)
        <self>: 42.29 KB (83.2%)
      inline-style-prefixer/lib/Prefixer.js|...: 8.58 KB (6.64%)
        <self>: 0 B (0.00%)
      inline-style-prefixer/lib/utils/getBrowserInformation.js|...: 4.41 KB (3.41%)
        <self>: 0 B (0.00%)
      inline-style-prefixer/lib/plugins/transition.js|...: 2.86 KB (2.22%)
        <self>: 0 B (0.00%)
      inline-style-prefixer/lib/Plugins.js|...: 1.46 KB (1.13%)
        <self>: 0 B (0.00%)
      exenv: 843 B (0.637%)
        <self>: 843 B (100%)
      <self>: 60.29 KB (46.6%)
    victory-animation: 48.52 KB (10.1%)
      d3-interpolate: 25.47 KB (52.5%)
        d3-color: 13.5 KB (53.0%)
          <self>: 13.5 KB (100%)
        <self>: 11.98 KB (47.0%)
      d3-ease: 5.23 KB (10.8%)
        <self>: 5.23 KB (100%)
      d3-timer: 2.77 KB (5.70%)
        <self>: 2.77 KB (100%)
      <self>: 15.05 KB (31.0%)
    radium/lib/resolve-styles.js|...: 11.91 KB (2.48%)
      <self>: 0 B (0.00%)
    victory-label: 10.11 KB (2.10%)
      <self>: 10.11 KB (100%)
    radium/lib/enhancer.js|...: 7.01 KB (1.46%)
      <self>: 0 B (0.00%)
    radium/lib/plugins/resolve-interaction-styles-plugin.js|...: 3.9 KB (0.812%)
      <self>: 0 B (0.00%)
    radium/lib/prefixer.js|...: 3.83 KB (0.797%)
      <self>: 0 B (0.00%)
    victory-util/lib/domain.js|...: 2.91 KB (0.605%)
      <self>: 0 B (0.00%)
    radium/lib/components/style-root.js|...: 2.8 KB (0.582%)
      <self>: 0 B (0.00%)
    radium/lib/components/style.js|...: 2.74 KB (0.571%)
      <self>: 0 B (0.00%)
    radium/lib/components/style-sheet.js|...: 2.72 KB (0.566%)
      <self>: 0 B (0.00%)
    victory-util/lib/style.js|...: 2.36 KB (0.491%)
      <self>: 0 B (0.00%)
    radium/lib/plugins/index.js|...: 1.54 KB (0.321%)
      <self>: 0 B (0.00%)
    victory-util/lib/index.js|...: 1.44 KB (0.299%)
      <self>: 0 B (0.00%)
    radium/lib/css-rule-set-to-string.js|...: 1.32 KB (0.275%)
      <self>: 0 B (0.00%)
    radium/lib/index.js|...: 1.18 KB (0.246%)
      <self>: 0 B (0.00%)
    radium/lib/keyframes.js|...: 1.13 KB (0.236%)
      <self>: 0 B (0.00%)
    radium/lib/append-important-to-each-value.js|...: 682 B (0.139%)
      <self>: 0 B (0.00%)
    radium/lib/get-state.js|...: 587 B (0.119%)
      <self>: 0 B (0.00%)
    radium/lib/plugins/prefix-plugin.js|...: 416 B (0.0845%)
      <self>: 0 B (0.00%)
    victory-label/lib/index.js|...: 90 B (0.0183%)
      <self>: 0 B (0.00%)
    <self>: 36.31 KB (7.55%)
  victory-line: 434.16 KB (16.3%)
    victory-util: 152.28 KB (35.1%)
      d3-scale: 114.49 KB (75.2%)
        d3-time-format: 27.13 KB (23.7%)
          <self>: 27.13 KB (100%)
        d3-format: 14.26 KB (12.5%)
          <self>: 14.26 KB (100%)
        d3-color: 13.5 KB (11.8%)
          <self>: 13.5 KB (100%)
        d3-arrays: 12.66 KB (11.1%)
          <self>: 12.66 KB (100%)
        d3-interpolate: 11.98 KB (10.5%)
          <self>: 11.98 KB (100%)
        d3-time: 10.38 KB (9.06%)
          <self>: 10.38 KB (100%)
        <self>: 24.59 KB (21.5%)
      reduce-css-calc: 6.01 KB (3.95%)
        reduce-function-call: 1.86 KB (30.9%)
          <self>: 1.86 KB (100%)
        balanced-match: 554 B (8.99%)
          <self>: 554 B (100%)
        <self>: 3.61 KB (60.1%)
      <self>: 31.77 KB (20.9%)
    radium: 111.93 KB (25.8%)
      inline-style-prefixer: 50.82 KB (45.4%)
        bowser: 8.52 KB (16.8%)
          <self>: 8.52 KB (100%)
        <self>: 42.29 KB (83.2%)
      exenv: 843 B (0.736%)
        <self>: 843 B (100%)
      <self>: 60.29 KB (53.9%)
    d3-shape: 50.53 KB (11.6%)
      d3-path: 4.57 KB (9.05%)
        <self>: 4.57 KB (100%)
      <self>: 45.96 KB (91.0%)
    victory-animation: 48.52 KB (11.2%)
      d3-interpolate: 25.47 KB (52.5%)
        d3-color: 13.5 KB (53.0%)
          <self>: 13.5 KB (100%)
        <self>: 11.98 KB (47.0%)
      d3-ease: 5.23 KB (10.8%)
        <self>: 5.23 KB (100%)
      d3-timer: 2.77 KB (5.70%)
        <self>: 2.77 KB (100%)
      <self>: 15.05 KB (31.0%)
    d3-shape/build/d3-shape.js|...: 45.96 KB (10.6%)
      <self>: 0 B (0.00%)
    <self>: 24.94 KB (5.74%)
  victory-pie: 387.98 KB (14.6%)
    victory-util: 152.28 KB (39.2%)
      d3-scale: 114.49 KB (75.2%)
        d3-time-format: 27.13 KB (23.7%)
          <self>: 27.13 KB (100%)
        d3-format: 14.26 KB (12.5%)
          <self>: 14.26 KB (100%)
        d3-color: 13.5 KB (11.8%)
          <self>: 13.5 KB (100%)
        d3-arrays: 12.66 KB (11.1%)
          <self>: 12.66 KB (100%)
        d3-interpolate: 11.98 KB (10.5%)
          <self>: 11.98 KB (100%)
        d3-time: 10.38 KB (9.06%)
          <self>: 10.38 KB (100%)
        <self>: 24.59 KB (21.5%)
      reduce-css-calc: 6.01 KB (3.95%)
        reduce-function-call: 1.86 KB (30.9%)
          <self>: 1.86 KB (100%)
        balanced-match: 554 B (8.99%)
          <self>: 554 B (100%)
        <self>: 3.61 KB (60.1%)
      <self>: 31.77 KB (20.9%)
    radium: 111.93 KB (28.8%)
      inline-style-prefixer: 50.82 KB (45.4%)
        bowser: 8.52 KB (16.8%)
          <self>: 8.52 KB (100%)
        <self>: 42.29 KB (83.2%)
      exenv: 843 B (0.736%)
        <self>: 843 B (100%)
      <self>: 60.29 KB (53.9%)
    d3-shape: 50.53 KB (13.0%)
      d3-path: 4.57 KB (9.05%)
        <self>: 4.57 KB (100%)
      <self>: 45.96 KB (91.0%)
    victory-animation: 48.52 KB (12.5%)
      d3-interpolate: 25.47 KB (52.5%)
        d3-color: 13.5 KB (53.0%)
          <self>: 13.5 KB (100%)
        <self>: 11.98 KB (47.0%)
      d3-ease: 5.23 KB (10.8%)
        <self>: 5.23 KB (100%)
      d3-timer: 2.77 KB (5.70%)
        <self>: 2.77 KB (100%)
      <self>: 15.05 KB (31.0%)
    <self>: 24.72 KB (6.37%)
  victory-bar: 373.13 KB (14.0%)
    victory-util: 152.28 KB (40.8%)
      d3-scale: 114.49 KB (75.2%)
        d3-time-format: 27.13 KB (23.7%)
          <self>: 27.13 KB (100%)
        d3-format: 14.26 KB (12.5%)
          <self>: 14.26 KB (100%)
        d3-color: 13.5 KB (11.8%)
          <self>: 13.5 KB (100%)
        d3-arrays: 12.66 KB (11.1%)
          <self>: 12.66 KB (100%)
        d3-interpolate: 11.98 KB (10.5%)
          <self>: 11.98 KB (100%)
        d3-time: 10.38 KB (9.06%)
          <self>: 10.38 KB (100%)
        <self>: 24.59 KB (21.5%)
      reduce-css-calc: 6.01 KB (3.95%)
        reduce-function-call: 1.86 KB (30.9%)
          <self>: 1.86 KB (100%)
        balanced-match: 554 B (8.99%)
          <self>: 554 B (100%)
        <self>: 3.61 KB (60.1%)
      <self>: 31.77 KB (20.9%)
    radium: 111.93 KB (30.0%)
      inline-style-prefixer: 50.82 KB (45.4%)
        bowser: 8.52 KB (16.8%)
          <self>: 8.52 KB (100%)
        <self>: 42.29 KB (83.2%)
      exenv: 843 B (0.736%)
        <self>: 843 B (100%)
      <self>: 60.29 KB (53.9%)
    victory-animation: 48.52 KB (13.0%)
      d3-interpolate: 25.47 KB (52.5%)
        d3-color: 13.5 KB (53.0%)
          <self>: 13.5 KB (100%)
        <self>: 11.98 KB (47.0%)
      d3-ease: 5.23 KB (10.8%)
        <self>: 5.23 KB (100%)
      d3-timer: 2.77 KB (5.70%)
        <self>: 2.77 KB (100%)
      <self>: 15.05 KB (31.0%)
    victory-util/lib/data.js|...: 10.63 KB (2.85%)
      <self>: 0 B (0.00%)
    victory-util/lib/prop-types.js|...: 4.89 KB (1.31%)
      <self>: 0 B (0.00%)
    victory-util/lib/scale.js|...: 3.48 KB (0.932%)
      <self>: 0 B (0.00%)
    <self>: 41.4 KB (11.1%)
  victory-scatter: 339.31 KB (12.8%)
    victory-util: 152.28 KB (44.9%)
      d3-scale: 114.49 KB (75.2%)
        d3-time-format: 27.13 KB (23.7%)
          <self>: 27.13 KB (100%)
        d3-format: 14.26 KB (12.5%)
          <self>: 14.26 KB (100%)
        d3-color: 13.5 KB (11.8%)
          <self>: 13.5 KB (100%)
        d3-arrays: 12.66 KB (11.1%)
          <self>: 12.66 KB (100%)
        d3-interpolate: 11.98 KB (10.5%)
          <self>: 11.98 KB (100%)
        d3-time: 10.38 KB (9.06%)
          <self>: 10.38 KB (100%)
        <self>: 24.59 KB (21.5%)
      reduce-css-calc: 6.01 KB (3.95%)
        reduce-function-call: 1.86 KB (30.9%)
          <self>: 1.86 KB (100%)
        balanced-match: 554 B (8.99%)
          <self>: 554 B (100%)
        <self>: 3.61 KB (60.1%)
      <self>: 31.77 KB (20.9%)
    radium: 111.93 KB (33.0%)
      inline-style-prefixer: 50.82 KB (45.4%)
        bowser: 8.52 KB (16.8%)
          <self>: 8.52 KB (100%)
        <self>: 42.29 KB (83.2%)
      exenv: 843 B (0.736%)
        <self>: 843 B (100%)
      <self>: 60.29 KB (53.9%)
    victory-animation: 48.52 KB (14.3%)
      d3-interpolate: 25.47 KB (52.5%)
        d3-color: 13.5 KB (53.0%)
          <self>: 13.5 KB (100%)
        <self>: 11.98 KB (47.0%)
      d3-ease: 5.23 KB (10.8%)
        <self>: 5.23 KB (100%)
      d3-timer: 2.77 KB (5.70%)
        <self>: 2.77 KB (100%)
      <self>: 15.05 KB (31.0%)
    <self>: 26.58 KB (7.83%)
  victory-chart: 291.88 KB (11.0%)
    victory-util: 152.28 KB (52.2%)
      d3-scale: 114.49 KB (75.2%)
        d3-time-format: 27.13 KB (23.7%)
          <self>: 27.13 KB (100%)
        d3-format: 14.26 KB (12.5%)
          <self>: 14.26 KB (100%)
        d3-color: 13.5 KB (11.8%)
          <self>: 13.5 KB (100%)
        d3-arrays: 12.66 KB (11.1%)
          <self>: 12.66 KB (100%)
        d3-interpolate: 11.98 KB (10.5%)
          <self>: 11.98 KB (100%)
        d3-time: 10.38 KB (9.06%)
          <self>: 10.38 KB (100%)
        <self>: 24.59 KB (21.5%)
      reduce-css-calc: 6.01 KB (3.95%)
        reduce-function-call: 1.86 KB (30.9%)
          <self>: 1.86 KB (100%)
        balanced-match: 554 B (8.99%)
          <self>: 554 B (100%)
        <self>: 3.61 KB (60.1%)
      <self>: 31.77 KB (20.9%)
    radium: 111.93 KB (38.3%)
      inline-style-prefixer: 50.82 KB (45.4%)
        bowser: 8.52 KB (16.8%)
          <self>: 8.52 KB (100%)
        <self>: 42.29 KB (83.2%)
      exenv: 843 B (0.736%)
        <self>: 843 B (100%)
      <self>: 60.29 KB (53.9%)
    <self>: 27.67 KB (9.48%)
  victory-label: 273.78 KB (10.3%)
    victory-util: 152.28 KB (55.6%)
      d3-scale: 114.49 KB (75.2%)
        d3-time-format: 27.13 KB (23.7%)
          <self>: 27.13 KB (100%)
        d3-format: 14.26 KB (12.5%)
          <self>: 14.26 KB (100%)
        d3-color: 13.5 KB (11.8%)
          <self>: 13.5 KB (100%)
        d3-arrays: 12.66 KB (11.1%)
          <self>: 12.66 KB (100%)
        d3-interpolate: 11.98 KB (10.5%)
          <self>: 11.98 KB (100%)
        d3-time: 10.38 KB (9.06%)
          <self>: 10.38 KB (100%)
        <self>: 24.59 KB (21.5%)
      reduce-css-calc: 6.01 KB (3.95%)
        reduce-function-call: 1.86 KB (30.9%)
          <self>: 1.86 KB (100%)
        balanced-match: 554 B (8.99%)
          <self>: 554 B (100%)
        <self>: 3.61 KB (60.1%)
      <self>: 31.77 KB (20.9%)
    radium: 111.93 KB (40.9%)
      inline-style-prefixer: 50.82 KB (45.4%)
        bowser: 8.52 KB (16.8%)
          <self>: 8.52 KB (100%)
        <self>: 42.29 KB (83.2%)
      exenv: 843 B (0.736%)
        <self>: 843 B (100%)
      <self>: 60.29 KB (53.9%)
    <self>: 9.57 KB (3.50%)
  <self>: 15.89 KB (0.598%)
react-treebeard: 1.48 MB (22.7%)
  velocity-react: 1.37 MB (92.4%)
    react: 629.34 KB (45.0%)
      fbjs: 32.93 KB (5.23%)
        <self>: 32.93 KB (100%)
      fbjs/lib/getMarkupWrap.js|...: 2.99 KB (0.475%)
        <self>: 0 B (0.00%)
      fbjs/lib/createNodesFromMarkup.js|...: 2.65 KB (0.421%)
        <self>: 0 B (0.00%)
      fbjs/lib/EventListener.js|...: 2.59 KB (0.412%)
        <self>: 0 B (0.00%)
      fbjs/lib/createArrayFromMixed.js|...: 2.3 KB (0.366%)
        <self>: 0 B (0.00%)
      fbjs/lib/toArray.js|...: 1.93 KB (0.307%)
        <self>: 0 B (0.00%)
      fbjs/lib/warning.js|...: 1.73 KB (0.275%)
        <self>: 0 B (0.00%)
      fbjs/lib/containsNode.js|...: 1.4 KB (0.222%)
        <self>: 0 B (0.00%)
      fbjs/lib/keyMirror.js|...: 1.24 KB (0.197%)
        <self>: 0 B (0.00%)
      fbjs/lib/camelizeStyleName.js|...: 1.01 KB (0.160%)
        <self>: 0 B (0.00%)
      fbjs/lib/hyphenateStyleName.js|...: 1005 B (0.156%)
        <self>: 0 B (0.00%)
      fbjs/lib/isTextNode.js|...: 628 B (0.0974%)
        <self>: 0 B (0.00%)
      <self>: 576.98 KB (91.7%)
    velocity-animate: 209.91 KB (15.0%)
      <self>: 209.91 KB (100%)
    react/lib/ReactDOMComponent.js|...: 36.08 KB (2.58%)
      <self>: 0 B (0.00%)
    react/lib/ReactMount.js|...: 35.91 KB (2.57%)
      <self>: 0 B (0.00%)
    react/lib/ReactClass.js|...: 27.17 KB (1.94%)
      <self>: 0 B (0.00%)
    react/lib/ReactCompositeComponent.js|...: 26.83 KB (1.92%)
      <self>: 0 B (0.00%)
    react/lib/SimpleEventPlugin.js|...: 17.03 KB (1.22%)
      <self>: 0 B (0.00%)
    react/lib/BeforeInputEventPlugin.js|...: 14.52 KB (1.04%)
      <self>: 0 B (0.00%)
    react/lib/ReactMultiChild.js|...: 14.34 KB (1.02%)
      <self>: 0 B (0.00%)
    react/lib/validateDOMNesting.js|...: 12.81 KB (0.915%)
      <self>: 0 B (0.00%)
    react/lib/ReactBrowserEventEmitter.js|...: 12.13 KB (0.867%)
      <self>: 0 B (0.00%)
    react/lib/ReactPropTypes.js|...: 12.06 KB (0.862%)
      <self>: 0 B (0.00%)
    react/lib/ChangeEventPlugin.js|...: 11.23 KB (0.803%)
      <self>: 0 B (0.00%)
    react/lib/ReactUpdateQueue.js|...: 10.68 KB (0.763%)
      <self>: 0 B (0.00%)
    react/lib/ReactElementValidator.js|...: 10.52 KB (0.752%)
      <self>: 0 B (0.00%)
    react/lib/ReactInstanceHandles.js|...: 10.4 KB (0.743%)
      <self>: 0 B (0.00%)
    react/lib/DOMProperty.js|...: 9.35 KB (0.668%)
      <self>: 0 B (0.00%)
    react/lib/Transaction.js|...: 9.33 KB (0.667%)
      <self>: 0 B (0.00%)
    react/lib/EventPluginHub.js|...: 9 KB (0.643%)
      <self>: 0 B (0.00%)
    react/lib/ReactUpdates.js|...: 8.34 KB (0.596%)
      <self>: 0 B (0.00%)
    react/lib/EventPluginRegistry.js|...: 8.21 KB (0.587%)
      <self>: 0 B (0.00%)
    react/lib/ReactElement.js|...: 7.89 KB (0.564%)
      <self>: 0 B (0.00%)
    react/lib/DOMPropertyOperations.js|...: 7.7 KB (0.550%)
      <self>: 0 B (0.00%)
    react/lib/HTMLDOMPropertyConfig.js|...: 7.45 KB (0.532%)
      <self>: 0 B (0.00%)
    react/lib/ReactEventListener.js|...: 7.34 KB (0.524%)
      <self>: 0 B (0.00%)
    react/lib/Danger.js|...: 6.8 KB (0.486%)
      <self>: 0 B (0.00%)
    react/lib/traverseAllChildren.js|...: 6.74 KB (0.481%)
      <self>: 0 B (0.00%)
    react/lib/ReactDOMSelection.js|...: 6.67 KB (0.476%)
      <self>: 0 B (0.00%)
    react/lib/EventPluginUtils.js|...: 6.63 KB (0.474%)
      <self>: 0 B (0.00%)
    react/lib/SelectEventPlugin.js|...: 6.55 KB (0.468%)
      <self>: 0 B (0.00%)
    react/lib/ReactDOMSelect.js|...: 5.96 KB (0.426%)
      <self>: 0 B (0.00%)
    react/lib/ReactChildren.js|...: 5.7 KB (0.407%)
      <self>: 0 B (0.00%)
    react/lib/ReactDOMInput.js|...: 5.61 KB (0.401%)
      <self>: 0 B (0.00%)
    react/lib/CSSPropertyOperations.js|...: 5.58 KB (0.399%)
      <self>: 0 B (0.00%)
    react/lib/EventPropagators.js|...: 5.1 KB (0.365%)
      <self>: 0 B (0.00%)
    react/lib/LinkedValueUtils.js|...: 5.06 KB (0.361%)
      <self>: 0 B (0.00%)
    react/lib/ReactComponent.js|...: 4.92 KB (0.352%)
      <self>: 0 B (0.00%)
    react/lib/DOMChildrenOperations.js|...: 4.88 KB (0.349%)
      <self>: 0 B (0.00%)
    react/lib/ReactReconcileTransaction.js|...: 4.47 KB (0.320%)
      <self>: 0 B (0.00%)
    react/lib/ReactChildReconciler.js|...: 4.42 KB (0.316%)
      <self>: 0 B (0.00%)
    react/lib/instantiateReactComponent.js|...: 4.41 KB (0.315%)
      <self>: 0 B (0.00%)
    react/lib/ReactDOMTextarea.js|...: 4.25 KB (0.303%)
      <self>: 0 B (0.00%)
    react/lib/ReactInputSelection.js|...: 4.22 KB (0.302%)
      <self>: 0 B (0.00%)
    react/lib/ReactDOMTextComponent.js|...: 4.22 KB (0.301%)
      <self>: 0 B (0.00%)
    react/lib/ReactDefaultInjection.js|...: 3.89 KB (0.278%)
      <self>: 0 B (0.00%)
    react/lib/ReactNoopUpdateQueue.js|...: 3.84 KB (0.275%)
      <self>: 0 B (0.00%)
    react/lib/EnterLeaveEventPlugin.js|...: 3.81 KB (0.273%)
      <self>: 0 B (0.00%)
    react/lib/SVGDOMPropertyConfig.js|...: 3.71 KB (0.265%)
      <self>: 0 B (0.00%)
    react/lib/ReactDOM.js|...: 3.62 KB (0.259%)
      <self>: 0 B (0.00%)
    react/lib/ReactReconciler.js|...: 3.47 KB (0.248%)
      <self>: 0 B (0.00%)
    react/lib/PooledClass.js|...: 3.46 KB (0.247%)
      <self>: 0 B (0.00%)
    react/lib/ReactOwner.js|...: 3.37 KB (0.241%)
      <self>: 0 B (0.00%)
    react/lib/ReactDOMFactories.js|...: 3.28 KB (0.234%)
      <self>: 0 B (0.00%)
    react/lib/setInnerHTML.js|...: 3.27 KB (0.234%)
      <self>: 0 B (0.00%)
    react/lib/ReactServerRendering.js|...: 3.22 KB (0.230%)
      <self>: 0 B (0.00%)
    react/lib/ReactDOMIDOperations.js|...: 3.22 KB (0.230%)
      <self>: 0 B (0.00%)
    react/lib/ReactNativeComponent.js|...: 2.95 KB (0.211%)
      <self>: 0 B (0.00%)
    react/lib/getEventKey.js|...: 2.86 KB (0.204%)
      <self>: 0 B (0.00%)
    react/lib/SyntheticKeyboardEvent.js|...: 2.7 KB (0.193%)
      <self>: 0 B (0.00%)
    react/lib/FallbackCompositionState.js|...: 2.43 KB (0.174%)
      <self>: 0 B (0.00%)
    react/lib/CallbackQueue.js|...: 2.38 KB (0.170%)
      <self>: 0 B (0.00%)
    react/lib/ReactRef.js|...: 2.29 KB (0.163%)
      <self>: 0 B (0.00%)
    react/lib/ReactServerRenderingTransaction.js|...: 2.25 KB (0.161%)
      <self>: 0 B (0.00%)
    react/lib/SyntheticMouseEvent.js|...: 2.15 KB (0.153%)
      <self>: 0 B (0.00%)
    react/lib/findDOMNode.js|...: 2.12 KB (0.152%)
      <self>: 0 B (0.00%)
    react/lib/ReactIsomorphic.js|...: 2.01 KB (0.143%)
      <self>: 0 B (0.00%)
    react/lib/EventConstants.js|...: 1.99 KB (0.142%)
      <self>: 0 B (0.00%)
    react/lib/SyntheticWheelEvent.js|...: 1.95 KB (0.139%)
      <self>: 0 B (0.00%)
    react/lib/isEventSupported.js|...: 1.93 KB (0.138%)
      <self>: 0 B (0.00%)
    react/lib/dangerousStyleValue.js|...: 1.88 KB (0.134%)
      <self>: 0 B (0.00%)
    react/lib/ReactDefaultBatchingStrategy.js|...: 1.87 KB (0.134%)
      <self>: 0 B (0.00%)
    react/lib/deprecated.js|...: 1.73 KB (0.123%)
      <self>: 0 B (0.00%)
    react/lib/accumulateInto.js|...: 1.7 KB (0.122%)
      <self>: 0 B (0.00%)
    react/lib/ReactEmptyComponent.js|...: 1.64 KB (0.117%)
      <self>: 0 B (0.00%)
    react/lib/ReactComponentEnvironment.js|...: 1.63 KB (0.117%)
      <self>: 0 B (0.00%)
    react/lib/flattenChildren.js|...: 1.61 KB (0.115%)
      <self>: 0 B (0.00%)
    react/lib/SyntheticUIEvent.js|...: 1.6 KB (0.115%)
      <self>: 0 B (0.00%)
    react/lib/React.js|...: 1.46 KB (0.104%)
      <self>: 0 B (0.00%)
    react/lib/ReactMarkupChecksum.js|...: 1.35 KB (0.0967%)
      <self>: 0 B (0.00%)
    react/lib/ReactInjection.js|...: 1.34 KB (0.0955%)
      <self>: 0 B (0.00%)
    react/lib/SyntheticTouchEvent.js|...: 1.3 KB (0.0930%)
      <self>: 0 B (0.00%)
    react/lib/ReactEventEmitterMixin.js|...: 1.27 KB (0.0909%)
      <self>: 0 B (0.00%)
    react/lib/DefaultEventPluginOrder.js|...: 1.23 KB (0.0880%)
      <self>: 0 B (0.00%)
    react/lib/ReactComponentBrowserEnvironment.js|...: 1.23 KB (0.0878%)
      <self>: 0 B (0.00%)
    react/lib/SyntheticClipboardEvent.js|...: 1.2 KB (0.0858%)
      <self>: 0 B (0.00%)
    react/lib/onlyChild.js|...: 1.18 KB (0.0845%)
      <self>: 0 B (0.00%)
    react/lib/setTextContent.js|...: 1.17 KB (0.0838%)
      <self>: 0 B (0.00%)
    react/lib/SyntheticCompositionEvent.js|...: 1.13 KB (0.0810%)
      <self>: 0 B (0.00%)
    react/lib/ReactBrowserComponentMixin.js|...: 1.12 KB (0.0802%)
      <self>: 0 B (0.00%)
    react/lib/SyntheticInputEvent.js|...: 1.12 KB (0.0799%)
      <self>: 0 B (0.00%)
    react/lib/SyntheticDragEvent.js|...: 1.1 KB (0.0786%)
      <self>: 0 B (0.00%)
    react/lib/SyntheticFocusEvent.js|...: 1.1 KB (0.0783%)
      <self>: 0 B (0.00%)
    react/lib/getTextContentAccessor.js|...: 994 B (0.0694%)
      <self>: 0 B (0.00%)
    react/lib/ReactMultiChildUpdateTypes.js|...: 861 B (0.0601%)
      <self>: 0 B (0.00%)
    react/lib/AutoFocusUtils.js|...: 816 B (0.0570%)
      <self>: 0 B (0.00%)
    react/lib/ReactDOMServer.js|...: 766 B (0.0535%)
      <self>: 0 B (0.00%)
    react/lib/quoteAttributeValueForBrowser.js|...: 746 B (0.0521%)
      <self>: 0 B (0.00%)
    react/lib/ReactPropTypeLocations.js|...: 549 B (0.0383%)
      <self>: 0 B (0.00%)
    react/lib/renderSubtreeIntoContainer.js|...: 463 B (0.0323%)
      <self>: 0 B (0.00%)
    react-addons-transition-group: 59 B (0.00412%)
      <self>: 59 B (100%)
    react/react.js|...: 56 B (0.00391%)
      <self>: 0 B (0.00%)
    <self>: 22.2 KB (1.59%)
  radium: 55.78 KB (3.68%)
    is-plain-object: 1.12 KB (2.01%)
      isobject: 283 B (24.6%)
        <self>: 283 B (100%)
      <self>: 866 B (75.4%)
    exenv: 843 B (1.48%)
      <self>: 843 B (100%)
    array-find: 478 B (0.837%)
      <self>: 478 B (100%)
    <self>: 53.37 KB (95.7%)
  shallowequal: 22.68 KB (1.50%)
    lodash.keys: 21.52 KB (94.9%)
      lodash.isarguments: 6.24 KB (29.0%)
        <self>: 6.24 KB (100%)
      lodash.isarray: 5.04 KB (23.4%)
        <self>: 5.04 KB (100%)
      lodash._getnative: 3.78 KB (17.6%)
        <self>: 3.78 KB (100%)
      <self>: 6.46 KB (30.0%)
    <self>: 1.16 KB (5.10%)
  babel-runtime: 12.01 KB (0.793%)
    core-js: 8.15 KB (67.9%)
      <self>: 8.15 KB (100%)
    <self>: 3.86 KB (32.1%)
  deep-equal: 3.8 KB (0.251%)
    <self>: 3.8 KB (100%)
  react-utils: 981 B (0.0633%)
    <self>: 981 B (100%)
  <self>: 19.76 KB (1.30%)
```

**Comment by ryan-roemer** [comment]
@bkniffler -- Thanks for the stats? Quick questions -- can you link / paste your webpack config? And you've got dedupe plugin enabled?

@boygirl -- Can you make sure we've got the same versions of the internal deps everywhere for this scenario? (Dedupe should collapse all of this stuff)

**Comment by bkniffler** [comment]
Hey, the plugins part of webpack looks like this:

```
plugins: [
         new webpack.ContextReplacementPlugin(/moment[\/\\]locale$/, /de/),
         new ExtractTextPlugin('bundle.css'),
         new webpack.DefinePlugin({
            "process.env": {
               BROWSER: JSON.stringify(true),
               NODE_ENV: JSON.stringify('production')
            }
         }),
         new webpack.optimize.DedupePlugin(),
         new webpack.optimize.OccurenceOrderPlugin(),
         new webpack.optimize.UglifyJsPlugin({
            compress: {
               warnings: false
            },
            output: {
               comments: false
            }
         })
      ],
```

**Comment by coopy** [comment]
Some stats:

* Simple component: https://github.com/FormidableLabs/victory-label
  * Takes 2.5 mins to install on my dual core MBP
  * 191.8 mb installed size (npm3)
  * minified bundle size: 94.2kb
* Complex component (contains line, axis, bar, etc): https://github.com/FormidableLabs/victory-chart
  * Takes 2.5 mins to install on my dual core MBP
  * 218.2 mb installed size (npm3)
  * minified bundle size: 290.8kb

We are working to optimize the bundle size. Another option is to only import the subcomponents you need (like `victory-line`) as in `import VictoryLine from "victory-chart/components/victory-line"`. This is currently blocked by https://github.com/FormidableLabs/victory-chart/issues/102.

**Comment by boygirl** [comment]
closing in favor of more specific issues #211 and #212
